notebooks/1_intelligent_agents_fall_2021/index.md

```diff
@@ -0,0 +1,194 @@
+    </ul>
+    <br>
+  <li><b>Discrete or continuous</b> (Are there a limited number of distinct, clearly defined states, percepts, and actions?)</li>  
+    <ul>
+        <li>We say an environment's state is discrete if there are a finite number of distinct states otherwise we say the environment's state in continuous.</li>
+         <li><b>examples</b>: Chess is discrete. Automatic driving is continuous.</li>
+    </ul>
+</ul>
+
+#### Types of environment example
+Here are a few examples of Identifying an environment's different dimensions.
+
+| environment| Fully observable? | Deterministic? |  Episodic? | Static? | Discrete?|Single agent?|
+| ----------- | ----------- | ----------- | ----------- | ----------- |  ----------- | ----------- |
+| Solitaire | No | Yes |  Yes | Yes | Yes | Yes |
+| Backgammon | Yes | No |  No | Yes | Yes | No |
+| Taxi driving | No | No |  No | No | No | No |
+| Medical diagnosis | No | No |  No | No | No | Yes |
+
+
+
+# Types of agents
+In this section we will introduce three basic kinds of basic agent programs.(The agent program is simply a program which implements the agent function.)
+<ul>
+  <li>Simple reflex agents</li>
+  <li>Goal-based agents</li>
+  <li>Utility-based agents</li>
+</ul>
+
+## Reflex agents
+This is the simplest kind of agent. They choose their next action only based on their current percept. In other words, they do not consider the future consequences of their actions and only consider <b>how the world IS.</b>  
+As an example look at this Pacman agent below, at each turn the agent look at its surrounding and chooses the direction that has a point in it and stops when there are no points around it.
+
+<img src="./images/reflex_agent.gif" width="500" style="margin-left: auto;margin-right: auto;"/>
+
+## Goal-based agents
+This kind of agent has a specific goal and it tries to reach that goal efficiently. They have a model of how the world evolves in response to actions, and they make decisions based on (hypothesized) consequences of actions to reach their goal state. Search and Planning are two subfields that are closely tied with these kinds of agents. In other words, these kinds of agents act on <b>how the world WOULD BE.</b>  
+as an example look at this Pacman agent below. the goal is to collect every point.
+
+<img src="./images/goal_based_agent.gif" width="500" style="margin-left: auto;margin-right: auto;"/>
+
+
+## Utility-based agents
+This kind of agent like goal-based agents has a goal. But they also have a Utility function they seek to reach their goal in a way that maximizes the utility function. For example, think about an automated car agent. They are many ways for this agent to get from point A to point B. But some of them are quicker, safer, cheaper. The utility function allows the agent to compare different states with each other and ask the question how happy am I in this state. 
```

Comment:
But they also have a **utility** function.
and use a period to end the sentence